packages/jest-jasmine2/src/jasmineAsyncInstall.ts

```diff
@@ -42,13 +42,22 @@ function promisifyLifeCycleFunction(
     if (!fn) {
       // @ts-expect-error: missing fn arg is handled by originalFn
       return originalFn.call(env);
+    } else if (typeof fn !== 'function') {
+      // Pass non-functions to Jest, which throws a nice error.
+      return originalFn.call(env, fn, timeout);
     }
 
-    const hasDoneCallback = typeof fn === 'function' && fn.length > 0;
+    const hasDoneCallback = fn.length > 0;
 
     if (hasDoneCallback) {
-      // Jasmine will handle it
-      return originalFn.call(env, fn, timeout);
+      // Give the function a name so it can be detected in call stacks, but
+      // otherwise Jasmine will handle it.
+      // @ts-expect-error: Support possible extra args at runtime
+      const asyncJestLifecycleWithCallback = function (this: any, ...args) {
```

Comment:
can we type `this` better? It should be `TestContext` or some such

Reply:
Done!